### ISSUE TITLE
Generated JAR not working

### ISSUE THREAD

**Issue by victorfuzaro** [Azure/azure-kusto-spark#191]
I am trying to compile the connector with its dependencies by running mvn assembly:single, the command works fine but the generated JAR doesn't. When I try to run a simple query, it returns me the following message:

`py4j.protocol.Py4JJavaError: An error occurred while calling o53.load.
: java.lang.ClassNotFoundException: Failed to find data source: com.microsoft.kusto.spark.datasource. Please find packages at http://spark.apache.org/third-party-projects.html`

I am using the latest connector version for Spark 2.4.

**Comment by ohadbitt** [comment]
hey 
Please provide us more details of what you are trying to do - what is the environment and what steps did you make

**Comment by victorfuzaro** [comment]
Hi ohadbitt,

I've just cloned the repo (the latest branch for spark 2.4) and ran the command: mvn assembly:single inside the connector folder. The proccess works fine but the generated jar file looks to be broken.

I am using a MacOS environment. if there is some specific information that you need please tell me which one.

**Comment by ohadbitt** [comment]
The question was - how you try to run it  ?

**Comment by victorfuzaro** [comment]
> The question was - how you try to run it ?

Hi ohadbitt, I will close this issue because I've got success for build the jar with through my Windows computer. I ran the command: mvn clean compile assembly:single and then it builded succesfully a working jar file. The same command fails to build on macOS. The command "mvn assembly:single" works fine for building, but the generated jar doesn't work and return the message that I mentioned before.

Thank you!
